presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -145,7 +155,7 @@ public PlanNodeDecorrelator(PlanNodeIdAllocator idAllocator, Lookup lookup)
         @Override
         public Optional<DecorrelationResult> visitLimit(LimitNode node, Void context)
         {
-            if (node.getCount() == 0) {
+            if (node.getCount() == 0 || node.isWithTies()) {
```

Comment:
can you please add a test for such queries?

Reply:
I needed to add the check to prevent wrong usage of this decorrelator. However, I can't create a test that would physically hit this path. This is because `LimitNode` `withTies` is rewritten before the decorrelation phase.

Reply:
I will assert that a correlated subquery containing `FETCH FIRST ... WITH TIES` fails (but this is due to WindowNode not decorrelating).